connection.js

```diff
@@ -744,155 +762,214 @@ function boardsizeSquareToDisplayString(boardsizeSquare) {
-    const argFamilySingularString = pluralFamilyStringToSingularString(argName);
-    // for example "speedsranked" -> "speed"
-    let argValueString = config[argName];
-    let notificationUnitConverted = notificationUnit;
-
-    if (argFamilySingularString.includes("boardsize")) {
-        argValueString = boardsizeSquareToDisplayString(config[argName]);
-        // for example boardsizeSquareToDisplayString("9,13,19"]) : "9x9, 13x13, 19x19"
-        notificationUnitConverted = boardsizeSquareToDisplayString(notificationUnit);
-    } else if (argFamilySingularString.includes("komi") && (notificationUnit === null)) {
-        notificationUnitConverted = "automatic";
-    }
-    conn_log(`${argFamilySingularString} ${rankedUnranked} `
-             + `-${notificationUnitConverted}-, not in -${argValueString}- `);
-    const msg = `${argFamilySingularString} -${notificationUnitConverted}- `
-                + `is not allowed on this bot ${rankedUnranked}, please `
-                + `choose one of these allowed ${argFamilySingularString}s `
-                + `${rankedUnranked}: -${argValueString}-`;
+function getAllowedFamiliesNotifToString(argName, notif) {
+    if (argName.includes("boardsizes")) {
+        return boardsizeSquareToDisplayString(notif);
+    }
+    if (argName.includes("komis") && (notif === null)) {
+        return "automatic";
+    } else {
+        return notif;
+    }
+}
+
+function getAllowedFamilyReject(argName, nameF, notif) {
+    const forRankedUnrankedGames = getForFromBLCRankedUnrankedGames("for ", "", argName, "");
+
+    const arg = config[argName];
+    const argToString = (argName.includes("boardsizes") ? boardsizeSquareToDisplayString(arg) : arg);
+    const notifToString = getAllowedFamiliesNotifToString(argName, notif);
+
+    conn_log(`${nameF} ${forRankedUnrankedGames} is ${notifToString}, not in ${argToString}.`);
+    const msg = `${nameF} ${notifToString} is not allowed on this bot${forRankedUnrankedGames}`
+                + `, please choose one of these allowed ${nameF}s${forRankedUnrankedGames}:\n${argToString}.`;
     return { reject: true, msg };
 }
 
-function familyObjectMIBL(familyName) {
-    let minMax = "";
-    let incDec = "";
-    let belAbo = "";
-    let lowHig = "";
-    const isMin = familyName.includes("min");
-    const isMax = familyName.includes("max");
+function getAllowedFamilyRejectResult(familyName, nameF, notif, notificationRanked) {
+    if (config[familyName] && !config[`${familyName}ranked`] && !config[`${familyName}unranked`] && !config[`allow_all_${familyName}`] && !config[`allowed_${familyName}`][notif]) {
+        return getAllowedFamilyReject(familyName, nameF, notif);
+    }
+    if (config[`${familyName}ranked`] && notificationRanked && !config[`allow_all_${familyName}_ranked`] && !config[`allowed_${familyName}_ranked`][notif]) {
+        return getAllowedFamilyReject(`${familyName}ranked`, nameF, notif);
```

Comment:
you gave me an idea with your comment, to avoid repetitions and mistakes, and also make it easier and nicer, we could add an extra line like

```Js
const [general, ranked, unranked] = getArgNamesGRU(familyName);
```